sql/core/src/test/scala/org/apache/spark/sql/parquet/ParquetSchemaSuite.scala

```diff
@@ -478,11 +478,12 @@ class ParquetSchemaSuite extends ParquetSchemaTest {
     StructType(Seq(
       StructField(
         "f1",
-        ArrayType(IntegerType, containsNull = true),
+        ArrayType(
+          StructType(Seq(StructField("num", IntegerType))), containsNull = false),
```

Comment:
Figured out why the test passes. It's because `array` is considered as a special name by Parquet LIST backwards-compatibility [rule 3] [1] (this is specific to parquet-avro). Whenever we see a 2nd level group type named `array`, Parquet format spec see it as the element type. That's why it's inferred as a struct here.

[1]: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules